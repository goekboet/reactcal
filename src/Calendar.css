.Calendar {

}

.Cell {
    width: 75px;
    height: 75px;
    border: 1px solid black;
    padding: 5px;
    float: left;
    opacity: 0.5;
}

.belong {
    opacity: 1;
}

.Cell:nth-child(7n+1) {
    clear: both;
}