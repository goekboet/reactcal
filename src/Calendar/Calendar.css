button {
    background-color: white;
    border: none;
    outline: none;
}

.navigator {
    text-align: center;
    padding-left: 12.5%;
}

.navigator button {
    font-size: 40px;
}

.navigator button:nth-child(1){
    width: 20%;
    text-align: left;
}

.navigator button:nth-child(2){
    width: 60%;
}

.navigator button:nth-child(3){
    width: 20%;
    text-align: right;
}

.viewSwitch {
    text-align: center;
    padding-bottom: 10px;
    padding-bottom: 20px;
    padding-left: 12.5%;
}

.viewSwitch button {
    font-size: 120%;
}

.viewSwitch span {
    margin: auto;
    
}

.calendar table {
    width: 100%;
    border-collapse: collapse;
    table-layout: fixed;
}

.calendar tr td:first-child {
    border: solid white;
    border-right: solid black;
    width: 50px;
    font-size: 100%;
    font-weight: 400;
}

.month tr {
    height: 80px;
}

.month td {
    border: solid black;
    font-size: 200%;
    font-weight: bold;
    text-align: center
}

.week td {
    border-right: solid gray 0.5px;
    border-bottom: dotted gray 0.2px;
}