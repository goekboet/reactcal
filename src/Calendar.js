import React from 'react';
import './Calendar.css';

const Calendar = (props) =>
{
    return (
    <div className="Calendar">
        <h1>{props.label}</h1>
        <div className="dayHeaders">
            {props.days.map((day, index) => 
                <div 
                    className={(day.belongs ? "belong" : "") + " Cell"} 
                    key={(index + 1).toString()}>
                    {day.date}
                </div>
            )
            }
        </div>
    </div>);
};

export default Calendar;