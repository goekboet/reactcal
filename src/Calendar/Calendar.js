import React, { Component } from 'react';
import Types from 'prop-types';
import Navigator from './Navigator/Navigator';
import ViewSwitch from './ViewSwitch/ViewSwitch';
import './Calendar.css'
import { Header, Body } from './TimeTable/TimeTable';
import { MonthLabel, WeekLabel, NextMonth, NextWeek, LastMonth, LastWeek, Hours, Headers, FirstOfMonth, Days } from './StateFuncs/StateFuncs';

/*
----------------------------------------------
Calendar

Denna komponent håller reda på en aktuell vy
och en aktuell tidpunkt. Med hjälp av lite logik
som bara är enkla anrop till date-fns biblioteket
renderar den en vy över den aktuella tidsperioden
och exponerar lite sätt till den att ändra aktuell
tidsperiod och tidpunkt.
----------------------------------------------
*/
class Calendar extends Component {
    constructor(props) {
        super(props);

        this.availiableViews = ["week", "month"];

        this.state = { 
            refDate: props.initDate,
            view: props.initView
        };
    }
    render() {
        return (
            <div className="calendar">
                <Navigator 
                    className="navigator"
                    currentViewLabel={this.label()}
                    now={() => this.setState({ refDate: new Date() })}
                    last={() => this.setState({ refDate: this.last() })}
                    next={() => this.setState({ refDate: this.next() })}/>
                <ViewSwitch
                    className="viewSwitch"
                    views={this.views()}
                    switch={v => () => this.setState({ view: v })}/>
                <table className={"currentView"}>
                    <Header headers={[''].concat(this.headers().map(x => x.label))}/>
                    <Body className={this.state.view} body={this.body()} />
                </table>
            </div>
            )
    }

    body() {
        return this.state.view === "week"
            ? Hours(this.headers())
            : Days(this.headers())
    }

    headers() {
        return this.state.view === "week" 
         ? Headers(this.state.refDate)
         : Headers(FirstOfMonth(this.state.refDate));
    }

    label() {
        return this.state.view === "week"
            ? WeekLabel(this.state.refDate)
            : MonthLabel(this.state.refDate);
    }

    nextView() {
        return this.state.view === "week"
            ? "month"
            : "week"
    }

    last() {
        return this.state.view === "week"
         ? LastWeek(this.state.refDate)
         : LastMonth(this.state.refDate)
    }
    
    next() {
        return this.state.view === "week"
            ? NextWeek(this.state.refDate)
            : NextMonth(this.state.refDate)
    }

    views() {
        return this.availiableViews.map(x => {
            return { view: x, current: this.state.view === x }
        })
    }
}

Calendar.propTypes = {
    initDate: Types.instanceOf(Date),
    initView: Types.oneOf(['week', 'month'])
}

export default Calendar;