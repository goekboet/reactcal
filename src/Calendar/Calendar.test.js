import React from 'react';
import Calendar from './Calendar';
import { create } from 'react-test-renderer';

it('renders week correctly', () => {
    const refdate = new Date(2017,12,24);
    const tree = create(<Calendar initDate={refdate} initView="week" />);
    expect(tree).toMatchSnapshot();
});

it('renders month correctly', () => {
    const refDate = new Date(2017,12,24);
    const tree = create(<Calendar initDate={refDate} initView="month" />);
    expect(tree).toMatchSnapshot();
})