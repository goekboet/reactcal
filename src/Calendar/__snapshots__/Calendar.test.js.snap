// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders month correctly 1`] = `
<div
  className="calendar"
>
  <div
    className="navigator"
  >
    <button
      onClick={[Function]}
    >
      ◀
    </button>
    <button
      onClick={[Function]}
    >
      January 2018
    </button>
    <button
      onClick={[Function]}
    >
      ▶
    </button>
  </div>
  <div
    className="viewSwitch"
  >
    <span>
      <button
        onClick={[Function]}
      >
        week
      </button>
      <button
        disabled={true}
        onClick={[Function]}
      >
        month
      </button>
    </span>
  </div>
  <table
    className="currentView"
  >
    <thead>
      <tr>
        <th>
          
        </th>
        <th>
          Mon
        </th>
        <th>
          Tue
        </th>
        <th>
          Wed
        </th>
        <th>
          Thu
        </th>
        <th>
          Fri
        </th>
        <th>
          Sat
        </th>
        <th>
          Sun
        </th>
      </tr>
    </thead>
    <tbody
      className="month"
    >
      <tr>
        <td>
          v.1
        </td>
        <td>
          1
        </td>
        <td>
          2
        </td>
        <td>
          3
        </td>
        <td>
          4
        </td>
        <td>
          5
        </td>
        <td>
          6
        </td>
        <td>
          7
        </td>
      </tr>
      <tr>
        <td>
          v.2
        </td>
        <td>
          8
        </td>
        <td>
          9
        </td>
        <td>
          10
        </td>
        <td>
          11
        </td>
        <td>
          12
        </td>
        <td>
          13
        </td>
        <td>
          14
        </td>
      </tr>
      <tr>
        <td>
          v.3
        </td>
        <td>
          15
        </td>
        <td>
          16
        </td>
        <td>
          17
        </td>
        <td>
          18
        </td>
        <td>
          19
        </td>
        <td>
          20
        </td>
        <td>
          21
        </td>
      </tr>
      <tr>
        <td>
          v.4
        </td>
        <td>
          22
        </td>
        <td>
          23
        </td>
        <td>
          24
        </td>
        <td>
          25
        </td>
        <td>
          26
        </td>
        <td>
          27
        </td>
        <td>
          28
        </td>
      </tr>
      <tr>
        <td>
          v.5
        </td>
        <td>
          29
        </td>
        <td>
          30
        </td>
        <td>
          31
        </td>
        <td>
          1
        </td>
        <td>
          2
        </td>
        <td>
          3
        </td>
        <td>
          4
        </td>
      </tr>
      <tr>
        <td>
          v.6
        </td>
        <td>
          5
        </td>
        <td>
          6
        </td>
        <td>
          7
        </td>
        <td>
          8
        </td>
        <td>
          9
        </td>
        <td>
          10
        </td>
        <td>
          11
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`renders week correctly 1`] = `
<div
  className="calendar"
>
  <div
    className="navigator"
  >
    <button
      onClick={[Function]}
    >
      ◀
    </button>
    <button
      onClick={[Function]}
    >
      v.4 January 2018
    </button>
    <button
      onClick={[Function]}
    >
      ▶
    </button>
  </div>
  <div
    className="viewSwitch"
  >
    <span>
      <button
        disabled={true}
        onClick={[Function]}
      >
        week
      </button>
      <button
        onClick={[Function]}
      >
        month
      </button>
    </span>
  </div>
  <table
    className="currentView"
  >
    <thead>
      <tr>
        <th>
          
        </th>
        <th>
          Mon
        </th>
        <th>
          Tue
        </th>
        <th>
          Wed
        </th>
        <th>
          Thu
        </th>
        <th>
          Fri
        </th>
        <th>
          Sat
        </th>
        <th>
          Sun
        </th>
      </tr>
    </thead>
    <tbody
      className="week"
    >
      <tr>
        <td>
          00:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          01:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          02:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          03:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          04:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          05:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          06:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          07:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          08:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          09:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          10:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          11:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          12:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          13:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          14:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          15:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          16:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          17:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          18:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          19:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          20:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          21:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          22:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
      <tr>
        <td>
          23:00
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
        <td>
          
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
